common/ferrostar/src/routing_adapters/osrm/algorithms.rs

```diff
@@ -0,0 +1,89 @@
+use super::models::AnyAnnotation;
+use crate::models::AnyAnnotationValue;
+use crate::routing_adapters::error::ParsingError;
+use serde_json::Value;
+use std::collections::BTreeMap;
+
+/// Get's the slice of annotations
+///
+/// Throws an [`ParsingError`] if the annotations are not present or the slice is out of bounds.
+pub(crate) fn get_annotation_slice(
+    annotations: Option<Vec<AnyAnnotationValue>>,
+    start_index: usize,
+    end_index: usize,
+) -> Result<Vec<AnyAnnotationValue>, ParsingError> {
+    annotations
+        .ok_or(ParsingError::Annotations {
+            error: "No annotations".to_string(),
+        })?
+        .get(start_index..end_index)
+        .ok_or(ParsingError::Annotations {
+            error: "Annotations slice index out of bounds ({start_index}..{end_index})".to_string(),
```

Comment:
Maybe show the size of the vec we're indexing into in the message for better debuggability?

I also feel like these error types probably belong in some other bucket. These aren't parsing errors.

Reply:
With regards to `ParsingError::Annotations`: The main goal of this error is to say: "Your API returned something out of spec for annotations" (with whatever reasoning). We're treating this different to normal parsing errors though, because we can easily still run a navigation session even if the annotations have to be tossed because they're malformed.